### PR TITLE
Added mock API implementation from backend and some temp design

### PR DESCRIPTION
**Pull request by xelleze**
The frontend design needs massive work I only created it for simple testing. 

The ticket I have didn't specifically say that I should design the front end so I wasn't sure

image

### CODE REVIEW

File: src/pages/VolunteerPage.jsx
Comment (MarioDanilo0111) [inline src/pages/VolunteerPage.jsx:1]: To improve the readability and  maintainability of the code, this `VolunteerPage` could be breaked down to smaller components for example ShelterList, FilterControls and BookingPopover.

File: src/pages/VolunteerPage.jsx [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+        setShowPopover(false);
+        setSelectedProduct(null);
+        setUserFirstName("");
+        setUserLastName("");
+        setFoundUser(null);
+        setFoundUserId(null);
+        setSearchError(null);
+    };
+
+    const handleBooking = async () => {
```
Comment (MarioDanilo0111) [inline src/pages/VolunteerPage.jsx:82]: After booking is confirmed, update the `avalibleShelters` state to reflect the reservation.
setting `places_left` to 0 or removing it if fully booked, ensuring the UI reflects the most recent availability.

**Review by MarioDanilo0111 — COMMENTED** [review]
I think that the code aligns well with the required criteria, and you've done a great job implementing the core functionality!
I noticed a couple of areas for potential improvement:
for instance, marking the UI feedback more clearly as 'reserved' could improve the user clarity.
I've also left a few additional comments in the code with suggestions. 
Great work overall!

File: src/pages/VolunteerPage.jsx
Comment (xelleze) [reply src/pages/VolunteerPage.jsx:1]: "The frontend design needs massive work I only created it for simple testing."

**Comment by xelleze** [comment]
The ticket didnt specificy the creating of the front end and I was unsure so I created it as a temp "The frontend design needs massive work I only created it for simple testing." but I will look inot it

**Review by jani-at-noq — APPROVED** [review]
Ok for now.

**Pull request merged**